fractal_server/app/runner/__init__.py

```diff
@@ -77,6 +78,10 @@ async def submit_workflow(
         exist, a new dataset with that name is created and within it a new
         resource with the same name.
     """
+    db_sync = next(DB.get_sync_db())
```

Comment:
For the record, what was the problem with the previous version (having db as an `AsyncSession` argument)?

Reply:
modifying the job object sparked a conflict with an open session in the endpoint and some subquery that is not compatible with async sessions. 

Now I make sure that all endpoint DB session be closed before registering the background task. This should guarantee that relevant objects are only present in one session at a time.